pkg/snapshot/restorer/restorer.go

```diff
@@ -90,15 +86,31 @@ func (r *Restorer) RestoreAndStopEtcd(ro brtypes.RestoreOptions, m member.Contro
 	return err
 }
 
-// Restore restore the etcd data directory as per specified restore options but returns the ETCD server that it statrted.
+// Restore restores the etcd data directory as per specified restore options but returns the ETCD server that it statrted.
 func (r *Restorer) Restore(ro brtypes.RestoreOptions, m member.Control) (*embed.Etcd, error) {
 	if err := r.restoreFromBaseSnapshot(ro); err != nil {
```

Comment:
Hmm you're right, this doesn't look right. I do remember a conversation a while back about eliminating full snapshots and using just delta snapshots from the initial bootstrapping of the etcd, so that the bucket will only have delta snapshots and no full snapshots. I don't fully remember why this discussion point came up in the first place.

But you're right, we currently don't have any use case where we need to try to restore from a bucket without a full snapshot. I'll change this behavior in this PR. Thanks for pointing it out.

Reply:
On second thought, it looks like there is already a negative test case for `restoring from only delta snapshots when no full snapshot is present` [here](https://github.com/gardener/etcd-backup-restore/blob/d5c31ac6c9fccae0aeac254cf2bf913c24530a3f/pkg/snapshot/restorer/restorer_test.go#L285-L332).

I will revert the change I made to this. Let's allow etcdbr to restore from only delta snapshots without a full snapshot, because I don't fully remember why we allow this. Maybe there was a usecase somewhere that I missed or forgot. Let's re-look at this in a different issue please.

Reply:
Raised an issue here: https://github.com/gardener/etcd-backup-restore/issues/611